### ISSUE TITLE
What do you mean by Python support?

### ISSUE THREAD

**Issue by retif** [MathItYT/mathlikeanim-rs#11]

**Comment by MathItYT** [comment]
Hello. I'm sorry for replying too late, I was busy with my university tasks...
This is a Rust library, and I'm planning to support Python through the PyO3 project in the future.

**Comment by retif** [comment]
It would be cool to reproduce Manim API there, which could use your framework under the hood and allow compilation to native/wasm.

**Comment by MathItYT** [comment]
Yeah, it would be cool. The problem is the compatibility between Manim stuff and WebAssembly.

- Manim depends on FFmpeg, and system commands can't be ran in the browser.
- Manim depends on OpenGL and Cairo libraries for Python, which aren't compatible with WebAssembly.
- Manim depends on system's LaTeX, which can't be emulated in the browser.
- Pango is also a problem.
- It depends on Skia for boolean operations, and that's also a problem.

A workaround on this is to fork Manim repository to make it compatible with Pyodide, and all names must be equal than in Manim so that it can be backwards compatible with a subset of Manim features.

**Comment by retif** [comment]
I think porting Manim itself to the web could give poor performance.

I meant to mimic Manim API as a thin wrapper over your rust library, like a python binding for the library. Then for the user this could give familiar interfaces to build with, but under the hood it would be way faster, both in wasm and native.

**Comment by guiferviz** [comment]
Would it be possible to use Python to write the code and then deploy an interactive version on the web, or would choosing Python limit us to running it only locally?

**Comment by MathItYT** [comment]
> Would it be possible to use Python to write the code and then deploy an interactive version on the web, or would choosing Python limit us to running it only locally?

Of course, it's possible. In fact, Plotly Python library uses sockets to interact with the JavaScript one, and that's a quite nice way to send data from Python to JavaScript.

**Comment by guiferviz** [comment]
OK, so as I understand it requires a Python backend, so it would not be possible to run it directly in a static web (like github pages for example), right?

**Comment by MathItYT** [comment]
Yeah, but you can deploy from sites like Railway

**Comment by guiferviz** [comment]
That could work. However, depending on the number of users, it may become resource-intensive. Nonetheless, it's a viable solution.

Another idea could be to translate Python code to JavaScript using tools like [Transcrypt](https://www.transcrypt.org/). This approach eliminates the need for a backend communicating via sockets. While it won't bring the full Python ecosystem to the web, it offers a potential solution for those who prefer using Python syntax in a web environment.

Thank you for your time! I look forward to forking the project and experimenting with the code.

**Comment by MathItYT** [comment]
Hey! I hope you had a good start to the year.

Python support is now ready :)

Now I'm closing this.
